pulsar-broker/src/main/java/org/apache/pulsar/broker/service/nonpersistent/NonPersistentRedeliveryTracker.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.broker.service.nonpersistent;
+
+import org.apache.bookkeeper.mledger.Position;
+import org.apache.pulsar.broker.service.RedeliveryTracker;
+
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class NonPersistentRedeliveryTracker implements RedeliveryTracker {
```

Comment:
Why is this tracker called "non-persistent"? Is it because the counter itself is not stored? 

In that case it's a bit confusing since there's the concept of non-persistent topics as well, which I think is orthogonal to the redelivery tracker.

Reply:
Because in the discuss of DLQ, has a solution is store the redelivery count in managed ledger, so i think it's possible to implement a PersistentRedeliveryTracker, so i named current implement to NonPersistentRedeliveryTracker.

Reply:
Sure, so then maybe we can rename into `InMemoryRedeliveryTracker` to avoid confusion with this be associated with `NonPersistentTopic`?

Reply:
Ok